plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -219,6 +220,37 @@ public static Table getIcebergTableWithMetadata(
         return new BaseTable(operations, quotedTableName(table), TRINO_METRICS_REPORTER);
     }
 
+    public static List<IcebergColumnHandle> getProjectedColumns(Schema schema, TypeManager typeManager)
+    {
+        ImmutableList.Builder<IcebergColumnHandle> projectedColumns = ImmutableList.builder();
+        StructType schemaAsStruct = schema.asStruct();
+        Map<Integer, NestedField> indexById = TypeUtil.indexById(schemaAsStruct);
+
+        for (Map.Entry<Integer, NestedField> entry : indexById.entrySet()) {
+            int fieldId = entry.getKey();
+            List<Integer> path = getPath(TypeUtil.indexParents(schemaAsStruct), fieldId);
```

Comment:
We don't need to calculate `TypeUtil.indexParents(schemaAsStruct)` for each entry in `indexById`